### PR TITLE
refactor: remove supply and borrow guard

### LINKED ISSUE #1628: Revert aave-v2 supply/borrow paused guards

### PR DESCRIPTION
**Pull request by pakim249CAL**
# Pull Request

## Issue(s) fixed

This pull request fixes #1628

### CODE REVIEW

File: src/aave-v2/EntryPositionsManager.sol [hunk 1]
```diff
@@ -194,12 +190,8 @@ contract EntryPositionsManager is IEntryPositionsManager, PositionsManagerUtils
         if (marketPauseStatus[_poolToken].isBorrowPaused) revert BorrowIsPaused();
 
         ERC20 underlyingToken = ERC20(market.underlyingToken);
-        DataTypes.ReserveConfigurationMap memory reserveConfig = pool.getConfiguration(
-            address(underlyingToken)
-        );
-
-        if (!reserveConfig.getBorrowingEnabled()) revert BorrowingNotEnabled();
-        if (reserveConfig.getFrozen()) revert FrozenOnPool();
+        if (!pool.getConfiguration(address(underlyingToken)).getBorrowingEnabled())
```
Comment (MerlinEgalite) [inline src/aave-v2/EntryPositionsManager.sol:193]: Should we keep this one after our discussion?

Comment (Rubilmax) [reply src/aave-v2/EntryPositionsManager.sol:193]: It seems we shouldn't?

Comment (MerlinEgalite) [reply src/aave-v2/EntryPositionsManager.sol:193]: That's what I understand on my side

Comment (MerlinEgalite) [reply src/aave-v2/EntryPositionsManager.sol:193]: Oh but no we need it because the asset could not be borrowable on Aave

Comment (pakim249CAL) [reply src/aave-v2/EntryPositionsManager.sol]: Yes I think I'm a bit confused because that check existed before upgrade-morpho-1

**Review by MathisGD — CHANGES_REQUESTED** [review]
I think that the `getBorrowingEnabled` was there before, and is actually very important ! (stETH)

**Comment by MathisGD** [comment]
>Oh but no we need it because the asset could not be borrowable on Aave

Just saw that, sorry

File: src/aave-v2/EntryPositionsManager.sol (hunk 1 above)
Comment (MerlinEgalite) [reply src/aave-v2/EntryPositionsManager.sol]: Ok let's close the PR then

**Comment by MathisGD** [comment]
I don't understand why you closed it ?

[Here](https://github.com/morpho-dao/morpho-v1/issues/1628) you explained me why we should remove them.

**Comment by MerlinEgalite** [comment]
> I don't understand why you closed it ?
> 
> [Here](https://github.com/morpho-dao/morpho-v1/issues/1628) you explained me why we should remove them.

Yes but:

> I think that the getBorrowingEnabled was there before, and is actually very important ! (stETH)

**Comment by Rubilmax** [comment]
From my understanding:

- `getBorrowingEnabled` is necessary because otherwise users could borrow from Morpho & borrow would revert on the pool but users could borrow 100% peer-to-peer and lenders wouldn't be able to withdraw (because pool would revert)

Note that aave governance can `setBorrowingEnabled` anytime
So we keep `isFrozen` check because the save governance have the exact same power over it and the impact would be similar

But then we should also consider behaving the same way on Morpho-Compound??

I genuinely don't understand our reasoning as a team here

**Comment by MerlinEgalite** [comment]
> From my understanding:
> 
> * `getBorrowingEnabled` is necessary because otherwise users could borrow from Morpho & borrow would revert on the pool but users could borrow 100% peer-to-peer and lenders wouldn't be able to withdraw (because pool would revert)
> 
> Note that aave governance can `setBorrowingEnabled` anytime So we keep `isFrozen` check because the save governance have the exact same power over it and the impact would be similar
> 
> But then we should also consider behaving the same way on Morpho-Compound??
> 
> I genuinely don't understand our reasoning as a team here

You're suggesting to do the borrow check on Morpho-Compound, correct?

**Comment by MerlinEgalite** [comment]
> > I think that the getBorrowingEnabled was there before, and is actually very important ! (stETH)
> 
> I'm talking about the isFrozen check. It seems (#1628) that a decisions has been made to remove those checks, and now we say we don't remove them (and even add the to Morpho-Compound). Both solutions are ok to me, but I don't understand our logic here.

Ok I think I asked to close the PR but in fact we should re-open it but only remove the frozen flags and keep the borrowing enabled flag because it's necessary

**Comment by Rubilmax** [comment]
> Ok I think I asked to close the PR but in fact we should re-open it but only remove the frozen flags and keep the borrowing enabled flag because it's necessary

Agreed that it is the logical conclusion we should have taken

Though I think we're doing something weird here as we are not behaving the same way whether the aave governance decides to `setBorrowingEnabled` or `setFrozen`, which have the same impact to me: we would allow people to borrow peer-to-peer and make lenders funds illiquid on Morpho

I know we have the same debate over & over but this is a new argument coming up: we're removing `isFrozen` but not `isBorrowingEnabled`

**Comment by MerlinEgalite** [comment]
> > Ok I think I asked to close the PR but in fact we should re-open it but only remove the frozen flags and keep the borrowing enabled flag because it's necessary
> 
> Agreed that it is the logical conclusion we should have taken
> 
> Though I think we're doing something weird here as we are not behaving the same way whether the aave governance decides to `setBorrowingEnabled` or `setFrozen`, which have the same impact to me: we would allow people to borrow peer-to-peer and make lenders funds illiquid on Morpho
> 
> I know we have the same debate over & over but this is a new argument coming up: we're removing `isFrozen` but not `isBorrowingEnabled`

`isBorrowingEnabled` is generally set at the asset listing event, so I think the behavior is a bit different between those 2 flags

**Review by Rubilmax — APPROVED** [review]

**Review by MerlinEgalite — APPROVED** [review]

**Review by Rubilmax — APPROVED** [review]

**Review by MathisGD — APPROVED** [review]

**Pull request merged**